File: app/src/main/java/com/fractalwrench/crazycats/common/PreConditions.java
package com.fractalwrench.crazycats.common;

public final class PreConditions {

    private PreConditions() {
        throw new AssertionError();
    }

    public static void checkNonNull(Object object) {
        if (object == null) {
            throw new IllegalArgumentException("Object cannot be null!");
        }
    }

}

File: app/src/main/java/com/fractalwrench/crazycats/common/PreConditions.kt
package com.fractalwrench.crazycats.common

class PreConditions private constructor() {

    init {
        throw AssertionError()
    }

    companion object {

        fun checkNonNull(`object`: Any?) {
            if (`object` == null) {
                throw IllegalArgumentException("Object cannot be null!")
            }
        }
    }

}
